### PR TITLE
Implemented HsmBookkeepingClient

### PR DESCRIPTION
**Pull request by kelvinator07**

### CODE REVIEW

**Review by jeremy-then — APPROVED** [review]
LGTM.

File: src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImplTest.java [hunk 1]
```diff
@@ -0,0 +1,458 @@
+package co.rsk.federate.signing.hsm.advanceblockchain;
```
Comment (nathanieliov) [inline src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImplTest.java:1]: Update this, it does not correspond to the path because you used a camel case for the package name. I think you should use lowercase only for the package name

File: src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImpl.java [hunk 2]
```diff
@@ -0,0 +1,200 @@
+package co.rsk.federate.signing.hsm.advanceblockchain;
```
Comment (nathanieliov) [inline src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImpl.java:1]: Please update the package name. Intellij is complaining cause the different cases of the word "blockchain".

File: src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImpl.java [hunk 3]
```diff
@@ -0,0 +1,200 @@
...
+
+    protected void sendBlockHeadersChunks(
+            List<String> blockHeaders,
+            String actualMethod,
+            boolean keepPreviousChunkLastItem
+    ) throws HSMClientException {
+        if (isStopped) {
+            return;
+        }
+        if (blockHeaders == null || blockHeaders.isEmpty()) {
```
[the remark this answers is not in the document]
Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImpl.java:77]: Theres a Test for that corner case incase we have another usage for that method

[the remark this answers is not in the document]
Comment (kelvinator07) [reply src/main/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImpl.java:77]: Theres a Test for that corner case incase we have another usage for that method

File: src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImplTest.java (hunk 1 above)
Comment (kelvinator07) [reply src/test/java/co/rsk/federate/signing/hsm/advanceBlockchain/HsmBookkeepingClientImplTest.java:1]: It has been updated but SonarCloud keeps complaining. Might be a bug from sonarcloud
